websockets/client.py

```diff
@@ -35,9 +35,8 @@ def write_http_request(self, path, headers):
 
         """
         self.path = path
-        self.request_headers = email.message.Message()
-        for name, value in headers:
-            self.request_headers[name] = value
+        self.request_headers = http.client.HTTPMessage()
+        self.request_headers._headers = headers     # HACK
```

Comment:
These two frequently used functions could also go on the class.

    set_header = lambda k, v: headers.append((k, v))
    get_header = lambda k: headers.get(k, '')